packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/queued_item_store/dart_queued_item_store.web.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:async';
+
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/queued_item_store/index_db/in_memory_queued_item_store.dart';
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/queued_item_store/index_db/indexed_db_adapter.dart';
+
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/queued_item_store/queued_item_store.dart';
+import 'package:aws_common/aws_common.dart';
+import 'package:drift/drift.dart';
+import 'package:meta/meta.dart';
+
+/// {@macro amplify_analytics_pinpoint_dart.dart_queued_item_store}
+class DartQueuedItemStore
+    with AWSDebuggable, AWSLoggerMixin
+    implements QueuedItemStore, Closeable {
+  /// {@macro amplify_analytics_pinpoint_dart.index_db_queued_item_store}
+  // ignore: avoid_unused_constructor_parameters
+  DartQueuedItemStore(QueryExecutor? driftQueryExecutor);
+
+  late final Future<QueuedItemStore> _database = () async {
+    if (await IndexedDbAdapter.checkIsIndexedDBSupported()) {
+      return IndexedDbAdapter();
+    } else {
+      logger.warn(
+        'IndexedDB is not available. '
+        'Falling back to in-memory storage.',
+      );
+      return InMemoryQueuedItemStore();
+    }
+  }();
+
+  @override
+  String get runtimeTypeName => 'DartQueuedItemStore';
+
+  @override
+  Future<void> add(String string) async {
+    final db = await _database;
+
+    try {
+      db.add(string);
+    } on Exception catch (e) {
+      _handleException(e);
+      rethrow;
+    }
+  }
+
+  @override
+  Future<void> deleteItems(Iterable<QueuedItem> items) async {
+    final db = await _database;
+
+    try {
+      db.deleteItems(items);
+    } on Exception catch (e) {
+      _handleException(e);
+      rethrow;
+    }
+  }
+
+  @override
+  Future<Iterable<QueuedItem>> get(int maxToGet) async {
+    final db = await _database;
+
+    try {
+      return db.get(maxToGet);
+    } on Exception catch (e) {
+      _handleException(e);
+      rethrow;
+    }
+  }
+
+  /// Clear IndexedDB data
+  /// Warning: The autoincrement of IndexedDB id is not reset
+  @override
+  @visibleForTesting
+  Future<void> clear() async {
+    final db = await _database;
+
+    try {
+      return db.clear();
+    } on Exception catch (e) {
+      _handleException(e);
+      rethrow;
+    }
+  }
+
+  void _handleException(Exception e) {
```

Comment:
I'm not sure I see the value in catching these just to add a log that isn't particularly helpful.